catalog/core/catalog-core-migratable/src/main/java/org/codice/ddf/catalog/migratable/impl/CatalogMigratableImpl.java

```diff
@@ -0,0 +1,221 @@
+     * @param config        export configuration information
+     */
+    public CatalogMigratableImpl(@NotNull String description, @NotNull CatalogFramework framework,
+            @NotNull FilterBuilder filterBuilder, @NotNull MigrationFileWriter fileWriter,
+            @NotNull CatalogMigratableConfig config) {
+        super(description, true);
+
+        notNull(framework, "CatalogFramework cannot be null");
+        notNull(filterBuilder, "FilterBuilder cannot be null");
+        notNull(fileWriter, "FileWriter cannot be null");
+        notNull(config, "Configuration object cannot be null");
+
+        this.framework = framework;
+        this.filterBuilder = filterBuilder;
+        this.fileWriter = fileWriter;
+        this.config = config;
+    }
+
+    /**
+     * Exports all the metacards currently stored in the catalog framework.
+     * <p>
+     * {@inheritDoc}
+     */
+    public MigrationMetadata export(Path exportPath) throws MigrationException {
+        taskManager = createTaskManager(config, createExecutorService(config));
+        config.setExportPath(exportPath.resolve(CATALOG_EXPORT_DIRECTORY));
+        fileWriter.createExportDirectory(config.getExportPath());
+
+        Collection<MigrationWarning> warnings = new ArrayList<>();
+        Map<String, Serializable> props = createMapWithNativeQueryMode();
+        Filter dumpFilter = filterBuilder.attribute(Metacard.ANY_TEXT)
+                .is()
+                .like()
+                .text("*");
+
+        QueryImpl exportQuery = createQuery(dumpFilter);
+        exportQuery.setPageSize(config.getExportQueryPageSize());
+        exportQuery.setRequestsTotalResultsCount(false);
+
+        QueryRequest exportQueryRequest = new QueryRequestImpl(exportQuery, props);
+
+        Subject subject = retrieveSystemSubject();
+        if (subject == null) {
+            LOGGER.error("System Subject was null");
+            throw new ExportMigrationException(DEFAULT_FAILURE_MESSAGE);
+        }
+
+        try {
+            subject.execute(new CatalogMigrationRunner(exportQuery, exportQueryRequest));
+        } catch (RuntimeException e) {
+            LOGGER.error("Internal error occurred when exporting catalog: {}", e.getMessage(), e);
+            throw new ExportMigrationException(DEFAULT_FAILURE_MESSAGE);
+        } finally {
+            taskManager.exportFinish();
+        }
+
+        return new MigrationMetadata(warnings);
+    }
+
+    // Factory method for creating QueryImpl objects so they can be mocked later.
+    QueryImpl createQuery(Filter dumpFilter) {
+        return new QueryImpl(dumpFilter);
+    }
+
+    // Factory method for creating MigrationTaskManager objects so they can be mocked later.
+    MigrationTaskManager createTaskManager(CatalogMigratableConfig config,
+            ExecutorService executorService) {
+        return new MigrationTaskManager(config, fileWriter, executorService);
```

Comment:
Same thing thats being done here https://github.com/codice/ddf/pull/559/files#diff-6eeb651045232ec710e12f35ee5b1d7dR145
Which I concur with you on.. But we're warping our classes solely for access of things in unit tests..